druntime/src/object.d

```diff
@@ -2535,8 +2535,28 @@ class Throwable : Object
      * $(D Throwable) is thrown from inside a $(D catch) block. The originally
      * caught $(D Exception) will be chained to the new $(D Throwable) via this
      * field.
+     *
+     * If the 0 bit is set in this, it means the next exception is refcounted,
+     * meaning this object owns that object and should destroy it on
+     * destruction.
+     *
+     * WARNING: This means we are storing an interior pointer to the next in
+     * chain, to signify that the next in the chain is actually reference
+     * counted. We rely on the fact that a pointer to a Throwable is not 1-byte
+     * aligned. It is important to use a local field to indicate reference
+     * counting since we are not allowed to look at GC-allocated references
+     * inside the destructor. It is very important to store this as a void*,
+     * since optimizers can consider an unaligned pointer to not exist.
      */
-    private Throwable   nextInChain;
+    private void*   _nextInChainPtr;
+
+    // 0 = not ref counted, 1 = ref counted
```

Comment:
just in case you're going to revise again once more - this comment seems superfluous now with the bool

Reply:
removed